custom/samveg/case_importer/operations.py

```diff
@@ -7,5 +12,6 @@ def run(cls, *args, **kwargs):
 class AddCustomCaseProperties(BaseOperation):
     @classmethod
     def run(cls, row_num, raw_row, fields_to_update):
-        # ToDo: Add new case property
+        today = datetime.utcnow().astimezone(pytz.timezone('Asia/Kolkata')).date()
```

Comment:
Is this is doing what you want? This will return a different result depending on the local timezone of the machine where it is running. For reference, it's 8:27am on my computer.
```py
>>> utcnow = datetime.utcnow()  # NOTE UTC timestamp (not local time)
>>> zones = [
...   'no conversion',
...   None,
...   pytz.timezone('EST'),
...   pytz.timezone('UTC'),
...   pytz.timezone('Asia/Kolkata'),
... ]
>>> for zone in zones:
...     if zone == 'no conversion':
...         print(utcnow, zone)
...     else:
...         print(utcnow.astimezone(zone), 'as', zone)
... 
2022-01-14 13:27:02.969647 no conversion
2022-01-14 13:27:02.969647-05:00 as None
2022-01-14 13:27:02.969647-05:00 as EST
2022-01-14 18:27:02.969647+00:00 as UTC
2022-01-14 23:57:02.969647+05:30 as Asia/Kolkata
```
Note +10:30 difference between `utcnow` and the Asia/Kolkata time (it should only be +5:30).

I think you want something like this
```py
>>> utcnow_zoned = utcnow.replace(tzinfo=pytz.timezone('UTC'))
>>> utcnow_asia = utcnow_zoned.astimezone(pytz.timezone('Asia/Kolkata'))
>>> print(utcnow_zoned); print(utcnow_asia)
2022-01-14 13:27:02.969647+00:00
2022-01-14 18:57:02.969647+05:30
```
From the [`astimezone` docs](https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone):

> If called without arguments (or with `tz=None`) the system local timezone is assumed for the target timezone.

The fact that we assume any timezone-unaware timestamp is UTC leads to trip-ups like this. /rant

Reply:
damn, thank you.

My intention here was to get the corresponding date that would be in `Asia/Kolkata` for when its run anywhere.
This would run on india.commcarehq.org which is hosted in India only, so I believe it would still get the right time.
But should definitely not rely on that, So this should be then datetime.now().astimezone(pytz.timezone('Asia/Kolkata')).date() then?

> The fact that we assume any timezone-unaware timestamp is UTC leads to trip-ups like this

Surely did

Reply:
> So this should be then datetime.now().astimezone(pytz.timezone('Asia/Kolkata')).date() then?

That still depends on server's local timezone, which I don't like to depend on because it might change or be different on different servers. However, on re-reading the `datetime` docs, I realized that `datetime.now()` accepts a timezone argument. This is what you want:

`datetime.now(pytz.timezone('Asia/Kolkata')).date()`

Reply:
https://github.com/dimagi/commcare-hq/pull/30921/commits/fba72d1124f7fb8878b06c23fb911a6f2a76a5ac